### PR TITLE
Make integration tests actually run.

### PR DESCRIPTION
**Pull request by dlorenc**
Also fix the pip test.

Unfortunately t.Parallel was making our tests only really run the last test case, a whole bunch of times simultaneously. We have a bunch of tests here that weren't really running.

Ref #183

### CODE REVIEW

**Comment by dlorenc** [comment]
OK, I'm going to disable the failing tests so we can get tests running again, then we should fix those in separate PRs and re-enable them.

File: pkg/cache/file_cache.go [hunk 1]
```diff
@@ -54,11 +55,15 @@ func (c *FileCache) HasLayer(layer types.BlobInfo) bool {
...
+	// Write the entry atomically. First write it to a .tmp name, then rename to the correct one.
+	f, err := ioutil.TempFile("", "")
 	if err != nil {
 		return nil, err
 	}
-	if _, err := io.Copy(entry, r); err != nil {
+	if _, err := io.Copy(f, r); err != nil {
+		return nil, err
+	}
+	if err := os.Rename(f.Name(), fullpath); err != nil {
```
Comment (nkubala) [inline pkg/cache/file_cache.go]: this is a change from another PR you have open right? not a huge fan of grouping this in with testing stuff, but if the tests fail without it then this is fine

Comment (dlorenc) [reply pkg/cache/file_cache.go]: Yeah I'll drop it from the other one. At the time I thought the flakiness was pretty minor, but actually running the tests makes it pretty clear that the flakiness is bad.

Comment (dlorenc) [reply pkg/cache/file_cache.go]: I thought about separating that out to it's own PR, but until tests are actually running we won't have much confidence in the PR itself...

Comment (nkubala) [reply pkg/cache/file_cache.go]: yeah that's kinda what I figured, just wanted to be sure

**Pull request merged**